### PR TITLE
Add performance testing workflow

### PR DESCRIPTION
**Pull request by rzats**
**Prerequisites**:

- [X] Unless it is a documentation hotfix it should be merged against the `dev` branch
- [X] Branch is up-to-date with the branch to be merged with, i.e. `dev`
- [X] Build is successful
- [X] Code is cleaned up and formatted

### Summary

Implements an automatable system for on-demand stability tests based on [the scalability-testing-rig PRD](https://docs.google.com/document/d/1znDZgeuciLIdm4-SEVWqcpllmEH0p8FE5r3ZWqUXj2U/edit#).

This first version of the testing rig launches a GitHub Actions runner off of a pull request based on a set comment (`/run performance test`) and, in order-
- sets up WireGuard in order to access the staging Delphi database
- clones the delphi-epidata repo, switching it to the pull request's branch
- sets up an environment with various other repos similar to the [install.sh](https://github.com/cmu-delphi/delphi-epidata/blob/dev/dev/local/install.sh) script
- builds and runs an instance of the epidata web server + API
- runs [Locust](https://locust.io/), a performance testing toolkit, targeting the local API we just created with a runtime of 15 minutes and 10 users spawning at a rate of 1 per second:
```
--users 10 --spawn-rate 1 --headless -t 15m
```
- links to Locust's report (which is contained in stdout) by creating a comment on the original pull request.

**IMPORTANT:** a prerequisite for this to function is for several secrets to be added to the repository.

These are as follows - various WireGuard credentials needed to set up the connection:
```
WG_PERF_ENDPOINT
WG_PERF_ENDPOINT_PUBLIC_KEY
WG_PERF_IPS
WG_PERF_ALLOWED_IPS
WG_PERF_PRIVATE_KEY
```
and the staging database connection string, including username & password:
```
DB_CONN_STRING
```

### CODE REVIEW

File: .github/workflows/performance-tests.yml [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+      - name: Set up repository # mimics install.sh in the README except that delphi is cloned from the PR rather than main
+        run: |
+          cd ..
+          mkdir -p driver/repos/delphi
+          cd driver/repos/delphi
+          git clone https://github.com/cmu-delphi/operations
+          git clone https://github.com/cmu-delphi/utils
+          git clone https://github.com/cmu-delphi/flu-contest
+          git clone https://github.com/cmu-delphi/nowcast
+          git clone https://github.com/cmu-delphi/github-deploy-repo
```
Comment (melange396) [inline .github/workflows/performance-tests.yml:44]: i think this can probably be removed?
```suggestion

```

**Review by melange396 — CHANGES_REQUESTED** [review]
looks pretty cool!  i had a couple little doc/naming things, but im mostly concerned with how much you might be able to take out of the repo cloning section.  `v4.py` is a straight copy out of the admin repository, correct?

**Comment by rzats** [comment]
@melange396 applied the fixes!

`v4.py` and the sample of queries are both a copy from the private repo, yeah. We can also upload the larger query sample file to S3 or similar, and then download it in the GitHub Action; or `git clone` the private admin repo although this may run into issues with token permissions.

**Review by melange396 — APPROVED** [review]
looks good!

**Pull request merged**